src/input/reader.go

```diff
@@ -0,0 +1,63 @@
+package input
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+)
+
+// UI is user-interface of input and output.
+type UI struct {
+	Interactive bool
+	Reader      io.ReadCloser
+	Writer      io.WriteCloser
+	once        sync.Once
+}
+
+var Default = &UI{}
+
+func (i *UI) init() {
+	tty, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
+	if err == nil {
+		i.Interactive = true
+		i.Reader = tty
+		i.Writer = tty
```

Comment:
I'd be surprised (and interested) if this is necessary. stdin and stdout should be the terminal already, in the general case, and that's all the bash/python version of this supported.

If you want to know if stdin/stdout is the terminal or not, there's a reasonably simple way to check: https://github.com/mattn/go-isatty/blob/master/isatty_linux.go or https://rosettacode.org/wiki/Check_input_device_is_a_terminal#Go (but it looks like nothing depends on `Interactive` anyway)